ws/public/main.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+(function () {
+    const canvas = document.getElementsByClassName('whiteboard')[0]
+    const context = canvas.getContext('2d')
+    const state = {
+        color: null,
+        drawing: false,
+        from: {},
+        to: {}
+    }
+    const socket = new WebSocket('ws://0.0.0.0:3000')
+
+    socket.onopen = function (e) {
+        console.log("[open] Connection established")
+    }
+
+    socket.onmessage = function (event) {
+        const message = JSON.parse(event.data)
+
+        if (message.name !== 'drawing') {
+            return
+        }
+
+        const canvasWidth = canvas.width
+        const canvasHeight = canvas.height
+        const data = {
+            from: {
+                x: message.data.from.x * canvasWidth,
+                y: message.data.from.y * canvasHeight
+            },
+            to: {
+                x: message.data.to.x * canvasWidth,
+                y: message.data.to.y * canvasHeight
+            },
+            color: message.data.color
+        }
+
+        drawLine(data.from, data.to, data.color)
+    }
+
+    const drawLine = (from, to, color) => {
+        context.beginPath()
+        context.moveTo(from.x, from.y)
+        context.lineTo(to.x, to.y)
+        context.strokeStyle = color
+        context.lineWidth = 2
+        context.stroke()
+        context.closePath()
+    }
+
+    const draw = (emit = false) => {
```

Comment:
unless I'm missing something, this function is only ever called with emit = true, so I'm not sure what this parameter is used for

Reply:
it was used but then i refactored the code and extracted the first part in a function